### PR TITLE
2.10.1 bump

### PR DESCRIPTION
**Pull request by mcg1969**
- Moved sha256 to top
- Removed noarch, restored s390x builds
- Updated python skip to 3.7
- Aligned dependencies to upstream setup.py
- Added setuptools and wheel to host dependencies
- Cleaned up metadata: reduced description, added doc_source_url
- Added python build dependencies to the metapackages so they build for the matrix

https://github.com/googleapis/python-api-core/blob/main/setup.py

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
 source:
   url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 97349cc18c2bb2415f64f1353a80273a289a61294ce3eb2f7ce682d251bdd997
+  sha256: {{ sha256 }}
 
 build:
   number: 0
-  noarch: python
-  skip: True      # [s390x]
+  # noarch: python
```
Comment (mcg1969) [inline recipe/meta.yaml:15]: We do not need to skip s390x for the base package—only for the `grcp` variants.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,35 +1,40 @@
 {% set name = "google-api-core" %}
-{% set version = "2.2.2" %}
+{% set version = "2.10.1" %}
+{% set sha256 = "e16c15a11789bc5a3457afb2818a3540a03f341e6e710d7f9bbf6cde2ef4a7c8" %}
 
 package:
-  name: {{ name|lower }}-split
+  name: {{ name }}
```
Comment (mcg1969) [inline recipe/meta.yaml:6]: There's no reason to use a "false" name at the top; the true name will do. That said, due to the following issue, it is necessary to use an explicit output with the same name:
https://github.com/conda/conda-build/issues/4023

File: recipe/meta.yaml [hunk 3]
```diff
@@ -52,13 +57,17 @@ outputs:
...
+        - pip check
 
   - name: {{ name }}-grpc
     build:
-      noarch: generic
+      # noarch: generic
+      skip: true  # [s390x]
     requirements:
+      host:
+        - python
```
Comment (mcg1969) [inline recipe/meta.yaml:68]: Adding `python` as a build requirement to the metapackages is necessary to ensure that the metapackages are built using the same build matrix as the main.

File: recipe/meta.yaml (hunk 1 above)
Comment (mcg1969) [reply recipe/meta.yaml:15]: We don't even need to skip for the grcp variants anymore. All platforms work now

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**